pennylane/qchem/integrals.py

```diff
@@ -127,21 +127,26 @@ def _fac2(n):
     return int(fac2(n) if n != -1 else 1)
 
 
-def _generate_params(params, args):
+def _generate_params(params, args, argnum=None):
     """Generate basis set parameters. The default values are used for the non-differentiable
     parameters and the user-defined values are used for the differentiable ones.
 
     Args:
         params (list(array[float])): default values of the basis set parameters
         args (list(array[float])): initial values of the differentiable basis set parameters
+        argnum (int | Sequence[int] | None): index (indices) of the positional argument(s) -
+            [``coordinates``, ``coeff``, ``alpha``] that should support differentiation. For
+            example, ``argnums=[0, 2]`` would mean derivatives can be computed with respect to both
+            ``coordinates`` and ``coeff``.
 
     Returns:
         list(array[float]): basis set parameters
     """
     basis_params = []
     c = 0
-    for p in params:
-        if p.requires_grad:
+    for i, p in enumerate(params):
+        # we iterate through argnum backwards because params order goes [alpha, coeff, r]
+        if getattr(p, "requires_grad", False) or (argnum is not None and 2 - i in argnum):
```

Comment:
You missed taking into account non-iterable `argnum` here.

Reply:
Oh oops.